publisher-sdk/src/main/java/com/criteo/publisher/DependencyProvider.java

```diff
@@ -598,30 +603,31 @@ public MetricSendingQueueConfiguration provideMetricSendingQueueConfiguration()
   public MetricRepository provideMetricRepository() {
     return getOrCreate(MetricRepository.class, new MetricRepositoryFactory(
         provideContext(),
-        provideMetricParser(),
+        provideJsonSerializer(),
         provideBuildConfigWrapper()
     ));
   }
 
-  @NonNull
-  public MetricParser provideMetricParser() {
-    return getOrCreate(MetricParser.class, () -> new MetricParser(
-        provideJsonSerializer()
-    ));
-  }
-
   @NonNull
   public JsonSerializer provideJsonSerializer() {
     return getOrCreate(JsonSerializer.class, () -> new JsonSerializer(
-        provideGson()
+        provideMoshi()
     ));
   }
 
   @NonNull
-  public Gson provideGson() {
-    return getOrCreate(Gson.class, () -> new GsonBuilder()
-        .registerTypeAdapterFactory(CustomAdapterFactory.create())
-        .create());
+  public Moshi provideMoshi() {
+    return getOrCreate(Moshi.class, () -> new Moshi.Builder()
+        .add(
+            RemoteLogLevel.class,
+            EnumJsonAdapter.create(RemoteLogLevel.class).withUnknownFallback(null).nullSafe()
+        )
+        .add(URI.class, new URIAdapter().lenient())
+        .add(URL.class, new URLAdapter().lenient())
+        .add(Boolean.class, new BooleanJsonAdapter().lenient().nullSafe())
+        .add(boolean.class, new BooleanJsonAdapter().lenient().nullSafe())
+        .add(ArrayListJsonAdapter.getFactory())
```

Comment:
I'm very curious why you need a so specific collection adapter, isn't mochi already handling lists ?